compiler/damlc/daml-doc/src/DA/Daml/Doc/Driver.hs

```diff
@@ -18,67 +21,99 @@ import Development.IDE.Types.Location
 import Development.IDE.Types.Diagnostics
 import Development.IDE.Types.Options
 
-import           Control.Monad.Extra
-import           Control.Monad.Except
+import Control.Monad.Extra
+import Control.Monad.Except
+import Data.Maybe
+import System.IO
+import System.Exit
+import System.Directory
+import System.FilePath
+
 import qualified Data.Aeson                        as AE
 import qualified Data.Aeson.Encode.Pretty          as AP
-import qualified Data.ByteString.Lazy              as BS
+import qualified Data.ByteString as BS
+import qualified Data.ByteString.Lazy as LBS
 import qualified Data.Text                         as T
 import qualified Data.Text.Encoding                as T
-import           System.Directory
-import           System.FilePath
 
+data DamldocOptions = DamldocOptions
+    { do_inputFormat :: InputFormat
+    , do_ideOptions :: IdeOptions
+    , do_outputPath :: FilePath
+    , do_outputFormat :: OutputFormat
+    , do_docTemplate :: Maybe FilePath
+    , do_docOptions :: [DocOption]
+    , do_inputFiles :: [NormalizedFilePath]
+    , do_docTitle :: Maybe T.Text
+    , do_combine :: Bool
+    }
 
 data InputFormat = InputJson | InputDaml
     deriving (Eq, Show, Read)
 
-damlDocDriver :: InputFormat
-              -> IdeOptions
-              -> FilePath
-              -> DocFormat
-              -> Maybe FilePath
-              -> [DocOption]
-              -> [NormalizedFilePath]
-              -> IO ()
-damlDocDriver cInputFormat ideOpts output cFormat prefixFile options files = do
+data OutputFormat = OutputJson | OutputHoogle | OutputDocs RenderFormat
+    deriving (Eq, Show, Read)
+
+-- | Run damldocs!
```

Comment:
Maybe calle it `runDamlDoc`s then? Not a big fan of calling things `driver`.

Reply:
👍